src/include/distributed/multi_logical_planner.h

```diff
@@ -189,15 +205,9 @@ extern bool SubqueryPushdown;
 extern MultiTreeRoot * MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 											  PlannerRestrictionContext *
 											  plannerRestrictionContext);
-extern bool JoinTreeContainsSubquery(Query *query);
-extern bool WhereClauseContainsSubquery(Query *query);
 extern bool FindNodeCheck(Node *node, bool (*check)(Node *));
 extern bool SingleRelationRepartitionSubquery(Query *queryTree);
-extern DeferredErrorMessage * DeferErrorIfCannotPushdownSubquery(Query *subqueryTree,
-																 bool
-																 outerMostQueryHasLimit);
 extern DeferredErrorMessage * DeferErrorIfUnsupportedUnionQuery(Query *queryTree);
```

Comment:
I think `DeferErrorIfUnsupportedUnionQuery` should go to pushdown planner as well. And, as far as I can see, `RecurringTuplesType` should go there as well (probably static in the file?).